### PR TITLE
Fix bed/door messages, add verification on use bed without house/PZ flag

### PR DESCRIPTION
**Pull request by ramon-bernardo**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
- Change `CANNOTUSETHISOBJECT` to `NOTPOSSIBLE` when trying to open house door.
- Show `YOUNEEDPREMIUMACCOUNT` if not premium when using bed.
- Show `YOUCANNOTUSETHISBED` if use bed without house. 

#3427 @emil92b

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

**Review by EPuncker — APPROVED** [review]

**Review by EPuncker — CHANGES_REQUESTED** [review]
can you add the orange message (talk type monster say) if a player "use" a bed outside a house tile?

`This bed can't be used, but Premium Account players can rent houses and sleep in beds there to regain health and mana.` related #2770

**Comment by ramon-bernardo** [comment]
Add this [commit](https://github.com/otland/forgottenserver/pull/3427/commits/e40ecfdde86459c4173e33fb952ae0e2bbfbb4af) to this PR seems interesting, we are dealing with house/bed.

File: src/bed.cpp [hunk 1]
```diff
@@ -95,7 +95,7 @@ BedItem* BedItem::getNextBedItem() const
 
 bool BedItem::canUse(Player* player)
 {
-	if (!player || !house || !player->isPremium()) {
+	if (!player || !house || !player->isPremium() || player->getZone() != ZONE_PROTECTION) {
```
Comment (EPuncker) [inline src/bed.cpp:98]: house tile is automatically protection zone so this check is not needed

Comment (ramon-bernardo) [reply src/bed.cpp:98]: This check is useless, yes. The player needs to be in PZ to use bed. In image above player can run from PVP if there is house without a wall.

**Review by EPuncker — APPROVED** [review]

**Review by DSpeichert — APPROVED** [review]

**Pull request merged**